tools/ncf.py

```diff
@@ -513,9 +522,20 @@ def generate_technique_content(technique, methods):
     # Add commentary for each new line in the metadata
     metadata_value = technique[metadata_key].replace("\n", "\n# ")
     content.append('# @'+ metadata_key +" "+ metadata_value)
+
+  bundle_param = ""
+  if technique["parameter"] > 0:
+    bundle_param = "("+", ".join([re.sub("\W","_",param["name"]) for param in technique["parameter"] ])+")" 
```

Comment:
Please use r"" or r'' insdeat of "" when writing a litteral regex.

Reply:
Be careful with \W
It doesn't have the same behaviour depending on the locale and on the ancoding on the string.
And the default encoding of the string is not the same in python 2 and python 3

I think we already have a canoncalization function in python somewhere